superset/security/manager.py

```diff
@@ -2645,7 +2645,7 @@ def has_guest_access(self, dashboard: "Dashboard") -> bool:
         dashboards = [
             r
             for r in user.resources
-            if r["type"] == GuestTokenResourceType.DASHBOARD.value
+            if r["type"] == GuestTokenResourceType.DASHBOARD
```

Comment:
@betodealmeida @sadpandajoe This have the potential to break current security checks but it's the correct comparison for the defined type. Let me know if it's ok or if we should change the type to `str` instead.